### PR TITLE
more config pruning and migrating

### LINKED ISSUE #83: [Refactor] Fix duplicate `config` and `examples` folder and update previous configs
In the past, the configs were all in [configs](https://github.com/OpenAccess-AI-Collective/axolotl/tree/main/configs). However, as things changed, some parts have been moved to the `examples` folder.

Furthermore, there are some old/invalid configs within the `configs` folder due to recent changes.

It would be good to move all configs to examples folder within each architecture for better maintenance and update those previous configs to work.

I'm curious if anyone has any better ideas?

### PR DESCRIPTION
**Pull request by winglian**
fixes #83 #174 #175

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,13 +16,14 @@
...
-| falcon  | ✅         | ✅                 | ✅  | ❌          | ❌                            | ❌               | ❓                  |
+|          | fp16/fp32 | fp16/fp32 w/ lora | qlora | 4bit-quant | 4bit-quant w/flash attention | flash attention | xformers attention |
+|----------|:----------|:------------------|------|------------|------------------------------|----------------|-----------------|
+| llama    | ✅         | ✅                 | ✅  | ✅          | ✅                            | ✅              | ✅               |
+| Pythia   | ✅         | ✅                 | ❓  | ❌          | ❌                            | ❌              | ❓               |
+| cerebras | ✅         | ✅                 | ❓  | ❌          | ❌                            | ❌              | ✅                |
+| mpt      | ✅         | ❌                 | ❓  | ❌          | ❌                            | ❌              | ❓               |
+| falcon   | ✅         | ✅                 | ✅  | ❌          | ❌                            | ❌              | ✅                |
+| gpt-j    | ✅         | ✅                 | ✅  | ❌          | ❌                            | ❓               | ✅                |
```
Comment (NanoCode012) [inline README.md]: should we add gpt Neo x? I did see that you deleted it though.

File: examples/gptj-qlora/config.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+base_model: EleutherAI/gpt-j-6b
```
Comment (NanoCode012) [inline examples/gptj-qlora/config.yml]: perhaps this should be moved into `gptj/config-qlora.yml` for `model/config-adapter.yml` format.

File: README.md (hunk 1 above)
Comment (winglian) [reply README.md]: yeah, gpt-neox needs some more time and love (technically, pythia is gpt-neox, but the gpt-neox-20b gives me headaches all day)

**Comment by NanoCode012** [comment]
Does qlora not work for other architectures due to some reason or untested?

**Comment by winglian** [comment]
> Does qlora not work for other architectures due to some reason or untested?

the matrix should have already been updated for qlora
Screenshot 2023-06-11 at 11 04 34 AM

**Pull request merged**